src/udev/udev-builtin-input_id.c

```diff
@@ -240,13 +241,18 @@ static bool test_pointers(struct udev_device *dev,
                         is_touchscreen = true;
         }
 
-        if (has_rel_coordinates && has_mouse_button)
-                is_mouse = true;
+        if (has_mouse_button)
+                if (has_rel_coordinates)
+                        is_mouse = true;
+                if (!has_abs_coordinates) /* exclude vmmouse and joysticks */
```

Comment:
Ah, like a game controller, not a touchpad. Indeed we call the latter `ID_INPUT_TOUCHPAD`. Can you please clarify that in a comment, or maybe rename `is_pad` to `is_game_controller` or perhaps more precisely `is_buttonsonly`?